### PR TITLE
Update to latest net9 preview

### LINKED ISSUE #2526: [Feature Request] M.IM conditionally targets .NET9-preview
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2343

Make sure the .NET9-preview is added to the GitHub action and the right builds. We should also have a build variable for .NET 9

start with 9.0.100-preview.2.24157.14
https://dotnet.microsoft.com/en-us/download/dotnet/9.0 and we'll update it every month. Using a * does not work well in this case.

An even better condition to add the new target framework is to modify `<TargetFrameworks>` conditionally (adding the line in green below)

```diff
<TargetFrameworks>net6.0; net7.0; net8.0; net462; net472; netstandard2.0</TargetFrameworks>
+ <TargetFrameworks Condition="'$(TargetNet9)'== 'True'">$(TargetFrameworks); net9.0</TargetFrameworks>
```

### PR DESCRIPTION
**Pull request by FuPingFranco**
## Description

Target latest NET9 preview-4.

Test Passing Builds:

- TargetNet9 = True (https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1309327)
- TargetNet9 = False (https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1309326)

Fixes #2526

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs [hunk 1]
```diff
@@ -198,7 +198,7 @@ internal bool ValidKeySize()
         /// <inheritdoc/>
         public override bool Sign(ReadOnlySpan<byte> input, Span<byte> signature, out int bytesWritten)
         {
-            if (input == null || input.Length == 0)
```
Comment (FuPingFranco) [inline src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs:201]: Comparing a span to 'null' might be redundant, 'null' literal will be implicitly converted to a 'Span<T>.Empty' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2265

File: src/Microsoft.IdentityModel.Tokens/SymmetricSignatureProvider.cs [hunk 2]
```diff
@@ -207,7 +207,7 @@ public override byte[] Sign(byte[] input)
         /// <inheritdoc/>
         public override bool Sign(ReadOnlySpan<byte> input, Span<byte> signature, out int bytesWritten)
         {
-            if (input == null || input.Length == 0)
```
Comment (FuPingFranco) [inline src/Microsoft.IdentityModel.Tokens/SymmetricSignatureProvider.cs:210]: Same as above

File: src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs:201]: interesting. thanks.

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Pull request merged**